### PR TITLE
Se master dev 3 delete workspace

### PR DESCRIPTION
**Pull request by SabineEmbacher**
Please review an merge

### CODE REVIEW

File: src/renderer/actions.ts [hunk 1]
```diff
@@ -41,6 +41,7 @@ import {
     isInputAssigned
 } from './containers/editor/value-editor-assign';
 import { ERROR_CODE_CANCELLED } from './webapi';
+import {DELETE_WORKSPACE_DIALOG_ID, OPEN_WORKSPACE_DIALOG_ID} from './containers/ChooseWorkspaceDialog';
```
Comment (forman) [inline src/renderer/actions.ts]:
```suggestion
import { DELETE_WORKSPACE_DIALOG_ID, OPEN_WORKSPACE_DIALOG_ID } from './containers/ChooseWorkspaceDialog';
```

File: src/renderer/actions.ts [hunk 2]
```diff
@@ -1043,10 +1063,21 @@ export function newWorkspaceInteractive() {
 
 function openRemoteWorkspace(dispatch: (action: (Action | ThunkAction)) => void,
                              getState: () => State) {
+    let state = getState();
```
Comment (forman) [inline src/renderer/actions.ts]: Unused local variable?

File: src/renderer/actions.ts [hunk 3]
```diff
@@ -1100,6 +1131,21 @@ export function openWorkspaceInteractive(): ThunkAction {
...
 
+/**
+ * Let user select a workspace, then ask whether to delete the selected workspace.
+ *
+ * @returns a Redux thunk action
+ */
+export function deleteWorkspaceInteractive(): ThunkAction {
+
+    return (dispatch: Dispatch, getState: GetState) => {
+        if (selectors.isLocalWebAPISelector(getState())) {
```
Comment (forman) [inline src/renderer/actions.ts]:
```suggestion
        if (!selectors.isLocalWebAPISelector(getState())) {
```

File: src/renderer/actions.ts [hunk 4]
```diff
@@ -1100,6 +1131,21 @@ export function openWorkspaceInteractive(): ThunkAction {
...
+        } else {
```
Comment (forman) [inline src/renderer/actions.ts]:
```suggestion
```

File: src/renderer/containers/ApplicationPage.tsx [hunk 5]
```diff
@@ -29,7 +29,7 @@ import { ViewManager, ViewRenderMap } from "../components/ViewManager";
 import { ViewLayoutState, ViewState, ViewPath, SplitDir } from "../components/ViewState";
 import { CSSProperties } from "react";
 import OperationStepDialog from "./OperationStepDialog";
-import ChooseWorkspaceDialog from './ChooseWorkspaceDialog';
+import ChooseWorkspaceDialog, {DELETE_WORKSPACE_DIALOG_ID, OPEN_WORKSPACE_DIALOG_ID} from './ChooseWorkspaceDialog';
```
Comment (forman) [inline src/renderer/containers/ApplicationPage.tsx]:
```suggestion
import ChooseWorkspaceDialog, { DELETE_WORKSPACE_DIALOG_ID, OPEN_WORKSPACE_DIALOG_ID } from './ChooseWorkspaceDialog';
```

File: src/renderer/containers/ChooseWorkspaceDialog.tsx [hunk 6]
```diff
@@ -5,10 +5,12 @@ import {connect, DispatchProp} from 'react-redux';
 import * as actions from '../actions';
 import * as selectors from '../selectors';
 import {ListBox, ListBoxSelectionMode} from '../components/ListBox';
+import {Checkbox} from '@blueprintjs/core';
```
Comment (forman) [inline src/renderer/containers/ChooseWorkspaceDialog.tsx]:
```suggestion
import { Checkbox } from '@blueprintjs/core';
```

File: src/renderer/containers/ChooseWorkspaceDialog.tsx [hunk 7]
```diff
@@ -5,10 +5,12 @@ import {connect, DispatchProp} from 'react-redux';
 import * as actions from '../actions';
 import * as selectors from '../selectors';
 import {ListBox, ListBoxSelectionMode} from '../components/ListBox';
```
Comment (forman) [inline src/renderer/containers/ChooseWorkspaceDialog.tsx]:
```suggestion
import { ListBox, ListBoxSelectionMode } from '../components/ListBox';
```

File: src/renderer/containers/ChooseWorkspaceDialog.tsx [hunk 8]
```diff
@@ -37,23 +42,36 @@ function mapStateToProps(state: State, ownProps: IChooseWorkspaceDialogOwnProps)
     }
     workspaceDir = isLocalWebAPI ? workspaceDir || '' : null;
     workspaceName = workspaceName || '';
+    let workspaceNames: string[];
+    if (state.data.workspace && state.data.workspaceNames) {
+        let workspaceBaseDir = state.data.workspace.baseDir;
```
Comment (forman) [inline src/renderer/containers/ChooseWorkspaceDialog.tsx]: Later, we want the API to return just the name in `state.data.workspace.baseDir`, so we don't need the following lines of code, right?

File: src/renderer/containers/ChooseWorkspaceDialog.tsx [hunk 9]
```diff
@@ -134,6 +170,23 @@ class ChooseWorkspaceDialog extends React.Component<IChooseWorkspaceDialogProps
             );
         }
 
+        let checkbox: any;
+        if (this.props.dialogId === DELETE_WORKSPACE_DIALOG_ID) {
+            checkbox =
+                // <div>
+                //     <label>
```
Comment (forman) [inline src/renderer/containers/ChooseWorkspaceDialog.tsx]: Remove out-commented code.

File: src/renderer/containers/ChooseWorkspaceDialog.tsx [hunk 10]
```diff
@@ -134,6 +170,23 @@ class ChooseWorkspaceDialog extends React.Component<IChooseWorkspaceDialogProps
...
+                <Checkbox label={'Delete entire workspace'}
+                          checked={this.state.deleteEntireWorkspace}
+                          onChange={this.handleCheckboxChange.bind(this)}
+                />
+            // Delete entire workspace
+            // </label>
+            // </div>
```
Comment (forman) [inline src/renderer/containers/ChooseWorkspaceDialog.tsx]: Remove out-commented code.

File: src/main/menu.ts [hunk 11]
```diff
@@ -11,6 +11,8 @@ const fileSubMenu: Array<any> = [
     actions.openWorkspace,
     actions.closeWorkspace,
     ________________,
+    actions.deleteWorkspace,
+    ________________,
```
Comment (forman) [inline src/main/menu.ts:12]: Maybe move to bottom?

**Review by forman — CHANGES_REQUESTED** [review]
Well done! Please address my change requests, then go on merging.

File: src/renderer/containers/ChooseWorkspaceDialog.tsx (hunk 8 above)
Comment (SabineEmbacher) [reply src/renderer/containers/ChooseWorkspaceDialog.tsx]: yes

**Pull request merged**